src/cpp/rtps/builtin/discovery/participant/PDPSimple.cpp

```diff
@@ -144,6 +144,49 @@ ParticipantProxyData * PDPSimple::createParticipantProxyData(
 {
     std::unique_lock<std::recursive_mutex> lock(*getMutex());
 
+    // decide if we dismiss the participant using the ParticipantFilteringFlags
+    const ParticipantFilteringFlags_t & flags = m_discovery.discovery_config.ignoreParticipantFlags;
+        
+    if(flags != ParticipantFilteringFlags_t::NO_FILTER)
+    {
+        const GUID_t & remote = participant_data.m_guid;
+        const GUID_t & local = getLocalParticipantProxyData()->m_guid;
+        
+        // filter if vendor or host are different
+        bool other_host = flags & ParticipantFilteringFlags::FILTER_DIFFERENT_HOST;
+        bool ignore = local.is_on_same_host_as(remote);
+
+        if( other_host && !ignore)
+        {
+            return nullptr;
+        }
```

Comment:
Changing this to the following code, we can remove the checks on `ignore` on the `else`
```
if(!local.is_on_same_host_as(remote))
{
    if(flags & ParticipantFilteringFlags::FILTER_DIFFERENT_HOST)
    {
        return nullptr;
    }
}
```